infrastructure/main.tf

```diff
@@ -1,3 +1,11 @@
+provider "azurerm" {}
+
+# Make sure the resource group exists
+resource "azurerm_resource_group" "rg" {
```

Comment:
You probably don't need to define the resource group explicitly here. You can retrieve it from the output of webapp module. If you decide to keep it here, it would be best to pass it to the webapp module, as well.